atlasdb-commons/src/test/java/com/palantir/async/initializer/AsyncInitializerTest.java

```diff
@@ -52,34 +57,100 @@ protected String getInitializingClassName() {
 
     @Test
     public void synchronousInitializationPropagatesExceptionsAndDoesNotRetry() throws InterruptedException {
-        AlwaysFailingInitializer initializer = new AlwaysFailingInitializer();
+        AsyncInitializer initializer = getMockedInitializer();
+
         assertThatThrownBy(() -> initializer.initialize(false))
                 .isInstanceOf(RuntimeException.class)
                 .hasMessage("Failed initializing");
-        Thread.sleep(WAIT_TIME_MILLIS);
-        assertThat(initializer.initializationAttempts).isEqualTo(1);
+        verify(initializer).tryInitialize();
+        verify(initializer, never()).scheduleInitialization();
     }
 
     @Test
     public void asyncInitializationCatchesExceptionAndRetries() {
+        AsyncInitializer initializer = getMockedInitializer();
+
+        initializer.initialize(true);
+        verify(initializer).tryInitialize();
```

Comment:
Could verify that this happens `times(2)`, or check `initializationAttempts` as below.